File: src/main/java/io/saso/dash/server/DashServerInitializer.java
package io.saso.dash.server;

import com.google.inject.Inject;
import com.google.inject.Provider;
import io.netty.channel.ChannelPipeline;
import io.netty.channel.socket.SocketChannel;
import io.netty.handler.codec.http.HttpObjectAggregator;
import io.netty.handler.codec.http.HttpServerCodec;

public class DashServerInitializer extends ServerInitializer
{
    private final Provider<ServerHttpHandler> handlerProvider;

    @Inject
    public DashServerInitializer(Provider<ServerHttpHandler> handlerProvider)
    {
        this.handlerProvider = handlerProvider;
    }

    @Override
    protected void initChannel(SocketChannel ch)
    {
        final ChannelPipeline pipeline = ch.pipeline();

        // initial http pipeline
        pipeline.addLast(new HttpServerCodec());
        pipeline.addLast("httpaggregator", new HttpObjectAggregator(65536));
        pipeline.addLast("httphandler", handlerProvider.get());
    }
}

File: src/main/kotlin/io/saso/dash/server/DashServerInitializer.kt
package io.saso.dash.server

import com.google.inject.Inject
import com.google.inject.Provider
import io.netty.channel.socket.SocketChannel
import io.netty.handler.codec.http.HttpObjectAggregator
import io.netty.handler.codec.http.HttpServerCodec
import io.saso.dash.server.handlers.ServerHttpHandler

public class DashServerInitializer
@Inject
constructor(val handlerProvider: Provider<ServerHttpHandler>):
        ServerInitializer()
{
    override fun initChannel(ch: SocketChannel)
    {
        val pipeline = ch.pipeline()

        // initial HTTP pipeline
        pipeline.addLast(HttpServerCodec())
        pipeline.addLast("httpaggregator", HttpObjectAggregator(65536))
        pipeline.addLast("httphandler", handlerProvider.get())
    }
}
